internal/workflow_test.go

```diff
@@ -0,0 +1,136 @@
+		WorkflowTaskTimeout:      3,
+		WaitForCancellation:      true,
+		WorkflowIDReusePolicy:    enums.WORKFLOW_ID_REUSE_POLICY_REJECT_DUPLICATE,
+		RetryPolicy:              newTestRetryPolicy(),
+		CronSchedule:             "todo",
+		Memo: map[string]interface{}{
+			"foo": "bar",
+		},
+		SearchAttributes: map[string]interface{}{
+			"foo": "bar",
+		},
+		ParentClosePolicy: enums.PARENT_CLOSE_POLICY_REQUEST_CANCEL,
+	}
+
+	// Require test options to have non-zero value for each field. This ensures that we update tests (and the
+	// GetChildWorkflowOptions implementation) when new fields are added to the ChildWorkflowOptions struct.
+	assertNonZero(t, opts)
+	// Check that the same opts set on context are also extracted from context
+	assert.Equal(t, opts, GetChildWorkflowOptions(WithChildWorkflowOptions(newTestWorkflowContext(), opts)))
+}
+
+func TestGetActivityOptions(t *testing.T) {
+	opts := ActivityOptions{
+		TaskQueue:              "foo",
+		ScheduleToCloseTimeout: time.Millisecond,
+		ScheduleToStartTimeout: time.Second,
+		StartToCloseTimeout:    time.Minute,
+		HeartbeatTimeout:       time.Hour,
+		WaitForCancellation:    true,
+		ActivityID:             "bar",
+		RetryPolicy:            newTestRetryPolicy(),
+	}
+
+	assertNonZero(t, opts)
+	assert.Equal(t, opts, GetActivityOptions(WithActivityOptions(newTestWorkflowContext(), opts)))
+}
+
+func TestGetLocalActivityOptions(t *testing.T) {
+	opts := LocalActivityOptions{
+		ScheduleToCloseTimeout: time.Minute,
+		StartToCloseTimeout:    time.Hour,
+		RetryPolicy:            newTestRetryPolicy(),
+	}
+
+	assertNonZero(t, opts)
+	assert.Equal(t, opts, GetLocalActivityOptions(WithLocalActivityOptions(newTestWorkflowContext(), opts)))
+}
+
+func TestConvertRetryPolicy(t *testing.T) {
+	someDuration := time.Minute
+	pbRetryPolicy := commonpb.RetryPolicy{
+		InitialInterval:        &someDuration,
+		MaximumInterval:        &someDuration,
+		BackoffCoefficient:     1,
+		MaximumAttempts:        2,
+		NonRetryableErrorTypes: []string{"some_error"},
+	}
+
+	assertNonZero(t, pbRetryPolicy)
+	// Check that converting from/to commonpb.RetryPolicy is transparent
+	assert.Equal(t, &pbRetryPolicy, convertToPBRetryPolicy(convertFromPBRetryPolicy(&pbRetryPolicy)))
+}
+
+func newTestWorkflowContext() Context {
+	return newWorkflowContext(&workflowEnvironmentImpl{
+		dataConverter: converter.GetDefaultDataConverter(),
+		workflowInfo: &WorkflowInfo{
+			Namespace:     "default",
+			TaskQueueName: "default",
+		},
+	}, nil, nil)
+}
+
+func newTestRetryPolicy() *RetryPolicy {
+	return &RetryPolicy{
+		InitialInterval:        1,
+		BackoffCoefficient:     2,
+		MaximumInterval:        3,
+		MaximumAttempts:        4,
+		NonRetryableErrorTypes: []string{"my_error"},
+	}
+}
+
+// assertNonZero checks that every top level value, struct field, and item in a slice is a non-zero value.
+func assertNonZero(t *testing.T, i interface{}) {
+	_assertNonZero(t, i, reflect.ValueOf(i).Type().Name())
+}
```

Comment:
Apologies if this is overwrought; I think it's nice to have peace of mind that tests would break if new fields are added but not accounted for in tests or conversion functions, but it's not essential to this changeset.